packages/cli/src/notices/notice_predictes_evaluator.ts

```diff
@@ -0,0 +1,184 @@
+  ) {}
+
+  evaluate = async (
+    notice: Notice,
+    opts: NoticesRendererParams,
+  ): Promise<boolean> => {
+    // If it's post deployment suppress notice by default. So that we show
+    // only notices that are allow listed for deployment.
+    let result = opts.event !== 'postDeployment';
+    for (const predicate of notice.predicates) {
+      switch (predicate.type) {
+        case 'nodeVersion':
+          if (!this.evaluateNodeVersion(predicate.versionRange)) {
+            return false;
+          }
+          break;
+        case 'osFamily':
+          if (!this.evaluateOsFamily(predicate.osFamily)) {
+            return false;
+          }
+          break;
+        case 'backendComponent':
+          // TODO out of scope for now due to complexity.
+          // Detecting backend components could be accomplished by:
+          // 1. Inspecting outputs file. (Not ideal, file is available only after successful deployment).
+          // 2. Inspecting cdk.out assembly (Does not require deployment, but relies on assembly format).
+          // 3. Have synthesis leave additional file into .amplify that we can read. (We control everything).
+          // 4. Have in memory ability to record information about verticals (Similar to 3 but requires integration with Toolkit to go first to spike it).
+          break;
+        case 'command':
+          if (!this.evaluateCommand(predicate.command)) {
+            return false;
+          }
+          break;
+        case 'errorMessage':
+          if (!this.evaluateErrorMessage(predicate.errorMessage, opts?.error)) {
+            return false;
+          }
+          break;
+        case 'frequency':
+          if (
+            !(await this.evaluateFrequency(
+              predicate.frequency,
+              opts.event,
+              notice.id,
+            ))
+          ) {
+            return false;
+          }
+          result = true;
+          break;
+        case 'validityPeriod':
+          if (!this.evaluateValidityPeriod(predicate.from, predicate.to)) {
+            return false;
+          }
+          break;
+        case 'packageVersion':
+          if (
+            !(await this.evaluatePackageVersion(
+              predicate.packageName,
+              predicate.versionRange,
+            ))
+          ) {
+            return false;
+          }
+          break;
+      }
+    }
+
+    return result;
+  };
+
+  private evaluateFrequency = async (
+    expectedFrequency: 'command' | 'deployment' | 'once' | 'daily',
+    event: NoticesRendererParams['event'],
+    noticeId: string,
+  ): Promise<boolean> => {
+    const projectName = await this.namespaceResolver.resolve();
+    const tracker = await this.noticesPrintingTrackerFile.read();
+    if (expectedFrequency === 'command') {
+      return event === 'postCommand' || event === 'listing';
+    } else if (expectedFrequency === 'deployment') {
+      return event === 'postDeployment' || event === 'listing';
```

Comment:
what's a `listing` event? Is it `notices --list`?

Reply:
Correct. perhaps `event` is wrong name. `purpose` might be better.
Any ideas ?

Reply:
I think `event` is fine since it seems to answer `when` rather than `where` or `why`. Maybe the `listing` should be `listingNoticesCommand`